internal/characters/barbara/barbara.go

```diff
@@ -0,0 +1,118 @@
+package barbara
+
+import (
+	"github.com/genshinsim/gcsim/pkg/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+)
+
+func init() {
+	core.RegisterCharFunc(keys.Barbara, NewChar)
+}
+
+type char struct {
+	*character.Tmpl
+	stacks     int
+	c6icd      int
+	skillInitF int
+	// burstBuffExpiry   int
+}
+
+func NewChar(s *core.Core, p core.CharacterProfile) (core.Character, error) {
+	c := char{}
+	t, err := character.NewTemplateChar(s, p)
+	if err != nil {
+		return nil, err
+	}
+	c.Tmpl = t
+	c.Energy = 80
+	c.EnergyMax = 80
+	c.Weapon.Class = core.WeaponClassCatalyst
+	c.BurstCon = 3
+	c.SkillCon = 5
+	c.NormalHitNum = 4
+
+	c.a1()
+
+	if c.Base.Cons >= 1 {
+		c.c1(1)
+	}
+	if c.Base.Cons >= 2 {
+		c.c2()
+	}
+	if c.Base.Cons >= 6 {
+		c.c6()
+	}
+	return &c, nil
+}
+
+func (c *char) a1() {
+	c.Core.AddStamMod(func(a core.ActionType) (float64, bool) { // @srl does this activate for the active char?
```

Comment:
stamina is on a per player basis so it doesnt matter what char is active. so this is fine